backend/src/db/cache_storage.py

```diff
@@ -34,28 +38,187 @@ def __init__(self):
             self.cache.set(key, value, expiration)
         else:
             self.cache.set(key, value, self.expiration_time)
 
-    def get_value(self, key):
+    def get_value(self, key: Any) -> Optional[Awaitable]:
+        """
+        Get the value associated with the given key from the cache.
+
+        Args:
+            key (Any): The key to retrieve the value for.
+
+        Returns:
+            Optional[Awaitable]: The value associated with the key, or None if key is not found.
+        """
         value = self.cache.get(key)
         if value is not None:
             return value
         return None
 
-    def delete_value(self, key):
+    def delete_value(self, key: Any) -> None:
+        """
+        Delete the key-value pair from the cache.
+
+        Args:
+            key (Any): The key to delete from the cache.
+
+        Returns:
+            None
+        """
         self.cache.delete(key)
 
-    def publish(self, channel, message):
-        pass
+    def publish_signal(
+        self,
+        pattern: bytes | str | memoryview,
+        serialized_data: bytes | memoryview | str | int | float,
+    ) -> Awaitable:
+        """
+        Publish a signal to a channel in the cache.
+
+        Args:
+            pattern (bytes | str | memoryview): The channel to publish the signal to.
+            serialized_data (bytes | memoryview | str | int | float): The data to publish.
+
+        Returns:
+            Awaitable
+        """
+        return self.cache.publish(channel=pattern, message=serialized_data)
+
+    def subscribe_signal(
+        self,
+        pattern: list,
+    ) -> None:
+        """
+        Subscribe to one or more channels to receive signals.
+
+        Args:
+            pattern (list): List of channels to subscribe to.
+
+        Returns:
+            None
+        """
+        self.pubsub.subscribe(*pattern)
+
+    def unsubscribe_signal(
+        self,
+        pattern: list,
+    ) -> None:
+        """
+        Unsubscribe from one or more channels.
+
+        Args:
+            pattern (list): List of channels to unsubscribe from.
+
+        Returns:
+            None
+        """
+        self.pubsub.unsubscribe(*pattern)
+
+    def get_signal(self) -> Optional[Union[None, dict[str, Union[None, str, bytes]]]]:
+        """
+        Get the next signal message from the subscribed channels.
+
+        Returns:
+            Optional[Union[None, dict[str, Union[None, str, bytes]]]]
+        """
+        while message := self.pubsub.get_message():
+            if message and message["type"] == "message":
+                return message
+            time.sleep(0.2)
+        return None
+
+
+class CacheHandler(ICacheHandler):
+    def __init__(self, redis: RedisStorage):
+        """
+        Initialize the CacheHandler with the provided RedisStorage instance.
+
+        Args:
+            redis (RedisStorage): The RedisStorage instance to use for caching.
+        """
+        self.redis = redis
```

Comment:
This is wrong. Firstly, parameter of storage shouldnt be focused on specific technology because what if we would switch to memcached or rabbitmq from redis ? The name would be misleading. You should call it like `cache_storage` and type it with interface instead of exact class.

Reply:
pool_storage maybe or store_storage, but pool name is better for me